### ISSUE TITLE
 stdout: error="read |0: file already closed

### ISSUE THREAD

**Issue by DavidHermes** [hashicorp/terraform#35667]
### Terraform Version

```shell
1.9.0 - 1.9.5 (Linux AMD64)
```

### Terraform Configuration Files

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.116.0"
    }
    random = {
      source  = "hashicorp/random"
      version = "3.3.1"
    }
  }

  backend "azurerm" {}
}

provider "azurerm" {
  subscription_id = var.subscription_id
  features {}
}

resource "azurerm_resource_group" "test" {

  name     = "TEST-RG"
  location = "North Europe"

}

provider "random" {}

### Debug Output

[ERROR] provider: error encountered while scanning stdout: error="read |0: file already closed

### Expected Behavior

Normal deployment

### Actual Behavior

This Error does not occur on every deployment but in about 30% of all cases.

Does not occur with terraform 1.8.5

### Steps to Reproduce

1.terraform init
2. terraform plan
3. terraform apply

### References

A simliar Issue;
https://github.com/hashicorp/go-plugin/issues/116

**Comment by jbardin** [comment]
Hi @DavidHermes,

That isn't a user-facing error from Terraform, are you experiencing a problem with Terraform planning or applying the configuration?

**Comment by DavidHermes** [comment]
It happens when applying a configuration.

**Comment by hingst** [comment]
I can confirm that this issue happens in some deployments. It seems to happen randomly, sometimes I get this error, but then re-running `terraform apply` with the same exact configuration works. Platform is Ubuntu x64.

This started happening after a recent Terraform update, but I can't pinpoint the exact version. It does not seem to be provider-related, as downgrading to an older Terraform version (1.6.x in my case) fixes it.

**Comment by jbardin** [comment]
Hello, the question we're waiting on is if there is an error reported by Terraform in the UI, or if Terraform is not behaving correctly. The log message above is not intended for the end user, it is probably an erroneous log message from an upstream library, but I'd like to rule out any possible Terraform issues before sending this over to other maintainers.

Thanks!

**Comment by jbardin** [comment]
Since we have not heard back in a while I'm going to close the issue. If you have any updates regarding the issue, feel free to open a new issue with the requested information. If you have more questions, you can also use [the community forum](https://discuss.hashicorp.com/c/terraform-core) where there are more people ready to help.

Thanks!
